File: pset1/cash.c
#include <stdio.h>
#include <math.h>
#include "cs50.h"

int main(void)
{
    float f = 0.0;
    do
    {
        f = get_float("Change owed: ");
    }
    while (f < 0);

    int cents = round(100 * f);
    int cnt = 0;
    int s[4] = {25, 10, 5, 1};
    while (cents)
    {
        for (int i = 0; i < 4; ++i)
        {
            if (cents >= s[i])
            {
                cents -= s[i];
                ++cnt;
                break;
            }
        }
    }
    printf("%d\n", cnt);

    return 0;
}

File: pset6/cash/cash.py
from cs50 import get_float


def main():
    change = getCash() * 100
    num = 0
    while change:
        if change >= 25:
            change -= 25
        elif change >= 10:
            change -= 10
        elif change >= 5:
            change -= 5
        else:
            change -= 1
        num += 1
    print(num)


def getCash():
    while True:
        money = get_float("Change owed: ")
        if money > 0:
            break
    return money


main()
